src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -80,16 +97,63 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
         catch (Exception ex)
         {
-            Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
-            await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace, cancellationToken);
+            if (ex is JsonException or CmsEventException)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
+                await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace,
+                    cancellationToken);
+            }
+
+            var deliveryAttempts = 0;
+
+            if (message.ApplicationProperties.TryGetValue(CustomMessageProperty, out object? attemptObj) &&
+                int.TryParse(attemptObj?.ToString(), out int existingAttempt))
+            {
+                deliveryAttempts = existingAttempt;
+            }
+
+            if (deliveryAttempts >= _maxMessageDeliveryAttempts)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
+                await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace,
+                    cancellationToken);
+            }
+            else
+            {
+                await RedeliverMessage(message, messageActions, deliveryAttempts, cancellationToken);
+            }
         }
     }
 
+    private async Task RedeliverMessage(ServiceBusReceivedMessage message, ServiceBusMessageActions messageActions,
+        int deliveryAttempts, CancellationToken cancellationToken)
+    {
+        var resubmittedMessage = new ServiceBusMessage()
+        {
+            ScheduledEnqueueTime = DateTime.UtcNow.AddSeconds(_messageDeliveryDelayInSeconds),
+            Subject = message.Subject,
+            Body = message.Body
+        };
+
+        var nextRetry = ++deliveryAttempts;
+
+        resubmittedMessage.ApplicationProperties.Add(CustomMessageProperty, nextRetry);
+
+        var sender = _serviceBusClient.CreateSender("contentful");
```

Comment:
This would be better if it was instantiated using `IAzureClientFactory<ServiceBusSender>` in the constuctor of the class; see the `ContentfulWebHook`.